app/Auth0/Auth0ServiceProvider.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Auth0;
+
+use Auth0\SDK\API\Management;
+use Auth0\SDK\Configuration\SdkConfiguration;
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
+use CultuurNet\UDB3\User\Auth0ManagementTokenGenerator;
+use CultuurNet\UDB3\User\Auth0ManagementTokenProvider;
+use CultuurNet\UDB3\User\Auth0UserIdentityResolver;
+use CultuurNet\UDB3\User\CacheRepository;
+use GuzzleHttp\Client;
+
+final class Auth0ServiceProvider extends AbstractServiceProvider
+{
+    protected function getProvidedServiceNames(): array
+    {
+        return [
+            'auth0.management-token',
+            Auth0UserIdentityResolver::class,
+        ];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared(
+            'auth0.management-token',
+            function () use ($container): string {
+                $provider = new Auth0ManagementTokenProvider(
+                    new Auth0ManagementTokenGenerator(
+                        new Client(),
+                        $container->get('config')['auth0']['client_id'],
+                        $container->get('config')['auth0']['domain'],
+                        $container->get('config')['auth0']['client_secret']
+                    ),
+                    new CacheRepository(
+                        $container->get('config')('auth0-management-token')
+                    )
+                );
+                return $provider->token();
+            }
+        );
+
+        $container->addShared(
+            'auth0.management-token',
```

Comment:
This seems to have the same name as the previous container entry.

Reply:
That should be `Auth0UserIdentityResolver::class` indeed